### PR TITLE
Feature/circular progress

### PR DESCRIPTION
**Pull request by dancasttro**
Creating Component Progress Indicator
https://natura.atlassian.net/browse/DSY-441

# Description

Please include a summary of the changes and features that were added. Also include which JIRA task is this related to (name and link). List any dependencies that were added or removed for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Snapshot

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/docs/src/components/CircularProgress/CircularProgress.story.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  />
+);
+
+export const CircularDeterminate = () => {
+  const [progress, setProgress] = React.useState(0);
+
+  React.useEffect(() => {
+    function tick() {
+      // reset when reaching 100%
+      setProgress(oldProgress => (oldProgress >= 100 ? 0 : oldProgress + 1));
```
Comment (robertLichtnow) [inline packages/docs/src/components/CircularProgress/CircularProgress.story.tsx]: This line could be
`setProgress(oldProgress => (oldProgress + 1) % 100);`

File: packages/docs/src/components/CircularProgress/CircularProgress.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+| size          |             'number' \| 'string'             |     40    |   The size of the circle. If using a number, the pixel unit is assumed. If using a string, you need to provide the CSS unit, e.g '3rem'.                                                                                                                                      |
+| thickness     |                     number                   |    3.6    |   The thickness of the circle.                                                                                                                                       |
+| value         |                     number                   |     0     |   The value of the progress indicator for the determinate and static variants. Value between 0 and 100.                                                                                                                                       |
+| variant       | 'indeterminate' \| 'determinate' \| 'static' | 'default' |   The variant to use. Use indeterminate when there is no progress value.                                                                                                                                       |
```
Comment (robertLichtnow) [inline packages/docs/src/components/CircularProgress/CircularProgress.docs.mdx]: Shouldn't `default` also be part of the `Type`

File: packages/docs/src/components/CircularProgress/CircularProgress.story.tsx (hunk 1 above)
Comment (robertLichtnow) [inline packages/docs/src/components/CircularProgress/CircularProgress.story.tsx]: This line could be
`setProgress(oldProgress => (oldProgress + 1) % 100);`

File: packages/docs/src/components/LinearProgress/LinearProgress.story.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        const diff = Math.random() * 10;
+        const diff2 = Math.random() * 10;
+        setCompleted(completed + diff);
+        setBuffer(completed + diff + diff2);
+      }
+    };
+  });
+
+  React.useEffect(() => {
+    function tick() {
```
Comment (vagnersaraiva) [inline packages/docs/src/components/LinearProgress/LinearProgress.story.tsx]: Change to arrow function

File: packages/docs/src/components/LinearProgress/LinearProgress.story.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    </div>
+  );
+};
+
+export const LinearDeterminate = () => {
+  const classes = useStyles();
+  const [completed, setCompleted] = React.useState(0);
+
+  React.useEffect(() => {
+    function progress() {
```
Comment (vagnersaraiva) [inline packages/docs/src/components/LinearProgress/LinearProgress.story.tsx]: Change to arrow function

File: packages/docs/src/components/CircularProgress/CircularProgress.story.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    variant="indeterminate"
+    disableShrink={boolean('DisabledShrink', false)}
+  />
+);
+
+export const CircularDeterminate = () => {
+  const [progress, setProgress] = React.useState(0);
+
+  React.useEffect(() => {
+    function tick() {
```
Comment (vagnersaraiva) [inline packages/docs/src/components/CircularProgress/CircularProgress.story.tsx]: Change to arrow function

File: packages/docs/src/components/CircularProgress/CircularProgress.docs.mdx (hunk 2 above)
Comment (dancasttro) [reply packages/docs/src/components/CircularProgress/CircularProgress.docs.mdx]: resolved

File: packages/docs/src/components/LinearProgress/LinearProgress.story.tsx (hunk 3 above)
Comment (dancasttro) [reply packages/docs/src/components/LinearProgress/LinearProgress.story.tsx]: Resolver

File: packages/docs/src/components/LinearProgress/LinearProgress.story.tsx (hunk 4 above)
Comment (dancasttro) [reply packages/docs/src/components/LinearProgress/LinearProgress.story.tsx]: Resolver

File: packages/docs/src/components/CircularProgress/CircularProgress.story.tsx (hunk 1 above)
Comment (dancasttro) [reply packages/docs/src/components/CircularProgress/CircularProgress.story.tsx]: Resolved

**Pull request merged**